### PR TITLE
chore(lint): unignore fixed rules

### PR DESCRIPTION
**Pull request by alexkrolick**
~~Ignored a few places that wanted to do reassignment through destructure, which looks like this:~~ changed the base config to ignore reassignment

```js
let { bar } = foo;
({ bar } = newFoo);
```

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @alexkrolick has signed the CLA. Thanks for the pull request!

File: .eslintrc.js [hunk 1]
```diff
@@ -7,7 +7,6 @@ module.exports = {
         'class-methods-use-this': 'off',
         'jsx-a11y/label-has-associated-control': 'off',
         'import/no-extraneous-dependencies': 'off', // fixme
-        'prefer-destructuring': 'off', // fixme
```
Comment (jstoffan) [inline .eslintrc.js]: Can we limit it to enforcing object destructuring only? That way we can continue using `items[0]` notation.

Comment (alexkrolick) [reply .eslintrc.js]: We can do that in `@box/frontend`, which this inherits from.

File: package.json [hunk 2]
```diff
@@ -201,7 +201,7 @@
         "ilib-tree-node": "^1.2.0",
         "immutable": "^3.7.4",
         "intl": "^1.2.5",
-        "intl-messageformat": "^2.2.0",
+        "intl-messageformat": "^7.2.1",
```
Comment (jstoffan) [inline package.json]: Probably want to revert this line. I saw that this version was throwing jest errors previously, which is why I left it at 2.2.0.

Comment (alexkrolick) [reply package.json]: will do

**Pull request merged**